src/vscode-bicep/src/utils/OutputChannelManager.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import {
+  createAzExtOutputChannel,
+  IAzExtOutputChannel,
+} from "@microsoft/vscode-azext-utils";
+
+import { Disposable } from "./disposable";
+
+export class OutputChannelManager extends Disposable {
```

Comment:
Not really seeing the need for OutputChannelManager.  Seems all it does is show the channel when logging.  If that's all, "AutoShowOutputChannel" or something would be better name.

But I'm not convinced that's acceptable to do.  I'd show it when we start deployment and when it completes or shows errors.  (Not during progress logging, although I understand that's not currently done.)

Reply:
I would like to keep this as is for below reasons:
- _azExtOutputChannel used here is disposable. Per suggestions from @shenglong 
(https://github.com/Azure/bicep/pull/6129#discussion_r825056135) OutputChannelManager inherits Disposable and registers _azExtOutputChannel. The OutputChannelManager instance created in extension.ts is registered as well
- Bicep Operations output channel is created in extension.ts and used in build, deploy and https://github.com/Azure/bicep/blob/0c6591c50c85a26e9b4508510068c053bc8a658e/src/vscode-bicep/src/tree/ResourceGroupTreeItem.ts#L114

Reply:
Btw, I was thinking about letting `OutputChannelManager` track the other `outputChannel` created in `extension.ts` as well, but that's a refactoring we can do in a separate PR.

Reply:
IAzExtOutputChannel also implements disposable.  It doesn't currently add anything.  My biggest concern though is this:

> But I'm not convinced that's acceptable to do. I'd show it when we start deployment and when it completes or shows errors. (Not during progress logging, although I understand that's not currently done.)

It's not good to keep focusing on the output window at random times.  That will be an issue later when you implement progress updates.

I guess it's up to you.

Reply:
Okay logged https://github.com/Azure/bicep/issues/6227 to further discuss this and make improvements. Thanks!